packages/victory-histogram/src/helper-methods.js

```diff
@@ -0,0 +1,245 @@
+import { assign, isNil } from "lodash";
+import { Helpers, LabelHelpers, Data, Domain, Scale } from "victory-core";
+import { getBarPosition } from "victory-bar";
+import isEqual from "react-fast-compare";
+import * as d3Array from "d3-array";
+import * as d3Scale from "d3-scale";
+
+const cacheLastValue = (func) => {
+  let called = false;
+  let lastArgs;
+  let lastReturnVal;
+
+  return (...args) => {
+    if (called && isEqual(lastArgs, args)) {
+      return lastReturnVal;
+    }
+
+    const value = func(...args);
+
+    called = true;
+    lastReturnVal = value;
+    lastArgs = args;
+
+    return value;
+  };
+};
+
+const dataOrBinsContainDates = ({ data, bins, x }) => {
+  const xAccessor = Helpers.createAccessor(x || "x");
+  const dataIsDates = data.some((datum) => xAccessor(datum) instanceof Date);
+  const binsHasDates = Array.isArray(bins) && bins.some((bin) => bin instanceof Date);
+
+  return dataIsDates || binsHasDates;
+};
+
+const getBinningFunc = ({ data, x, bins, dataOrBinsContainsDates }) => {
+  const xAccessor = Helpers.createAccessor(x || "x");
+  const bin = d3Array.bin().value(xAccessor);
+  const niceScale = (dataOrBinsContainsDates ? d3Scale.scaleTime() : d3Scale.scaleLinear())
+    .domain(d3Array.extent(data, xAccessor))
+    .nice();
+
+  if (Array.isArray(bins)) {
+    bin.domain([bins[0], bins[bins.length - 1]]);
+    bin.thresholds(bins);
+
+    return bin;
+  }
+
+  if (Number.isInteger(bins)) {
+    bin.domain(niceScale.domain());
+    bin.thresholds(bins);
+
+    return bin;
+  }
+
+  if (dataOrBinsContainsDates) {
+    bin.domain(niceScale.domain());
+    bin.thresholds(niceScale.ticks());
+
+    return bin;
+  }
+
+  bin.domain(niceScale.domain());
+
+  return bin;
+};
+
+export const getFormattedData = cacheLastValue(({ data = [], x, bins }) => {
+  if ((!data || !data.length) && !Array.isArray(bins)) {
+    return [];
+  }
+  const dataOrBinsContainsDates = dataOrBinsContainDates({ data, bins, x });
+  const binFunc = getBinningFunc({ data, x, bins, dataOrBinsContainsDates });
+  const binnedData = binFunc(data).filter(({ x0, x1 }) => x0 !== x1);
+
+  const formattedData = binnedData.map((bin) => ({
+    x0: dataOrBinsContainsDates ? new Date(bin.x0) : bin.x0,
```

Comment:
using _x0 and _x1 was causing positioning issues, so I went with x0 and x1

Reply:
Is this because of `Helpers.scalePoint` in `Bar`?